collection+subdoc.go

```diff
@@ -145,6 +145,30 @@ func evalSubdocPath(subdoc any, path []string) (any, error) {
 	return subdoc, nil
 }
 
+// Upserts the value at the specified JSON path in the source.
+// Returns ErrPathMismatch if non-leaf path entries do not exist.
+func upsertSubdocValue(source any, path []string, value interface{}) error {
+
+	// eval path exists
+	subdoc, err := evalSubdocPath(source, path[0:len(path)-1])
+	if err != nil {
+		return err
+	}
+
+	parent, ok := subdoc.(map[string]any)
```

Comment:
Can we map this a compile time error by requiring map[string]any in compiler string?

Reply:
This is required to support non-map JSON.